### ISSUE TITLE
Add support for @NotEmpty annotation

### ISSUE THREAD

**Issue by kog** [mbknor/mbknor-jackson-jsonSchema#56]
Looking at the [Bean Validation spec](http://beanvalidation.org/2.0/spec/#builtinconstraints-notempty), `@NotEmpty` states:

```java
/**
 * The annotated element must not be {@code null} nor empty. Supported types are:
 * <ul>
 * <li>{@code CharSequence} (length of character sequence is evaluated)</li>
 * <li>{@code Collection} (collection size is evaluated)</li>
 * <li>{@code Map} (map size is evaluated)</li>
 * <li>Array (array length is evaluated)</li>
 * </ul>
 *
 * @author Emmanuel Bernard
 * @author Hardy Ferentschik
 *
 * @since 2.0
 */
```

It looks like this would affect JSON Strings, Objects, and Arrays:

- The property should be marked as [required](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.4.3)
- For Strings, a [minLength](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.2.2) of `1` would be generated for the property
- For Arrays, a [minItems](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.3.3) of `1` would be generated for the property
- For Objects, a [minProperties](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.4.2) of `1` would be generated for the property

For other types we can log something like "property X is marked `@NotEmpty`, but this does not apply to [type name], ignoring..."

**Comment by kog** [comment]
I've got a bit of prototypical code for this that I'll get going into a PR from my forked repo soon (hopefully this weekend?).

**Comment by mbknor** [comment]
Looking forward to it getting the PR :)

**Comment by kog** [comment]
Yeah, sorry, it turns out my Scala is not as great as it could be. I've got `@NotBlank` going first and am just refactoring things. I can submit it for a pull request if you'd like to suggest some stuff ;)

**Comment by mbknor** [comment]
Closed in https://github.com/mbknor/mbknor-jackson-jsonSchema/pull/59

**Comment by kog** [comment]
Er, #59 added `@NotBlank`, this is for `@NotEmpty`. You mind re-opening?

**Comment by kog** [comment]
Oh, I see... for some dumb reason I typo'd 59 into my git log for the last pull request. Doh. Oh well, PR is up for NotEmpty.

**Comment by mbknor** [comment]
Closed in https://github.com/mbknor/mbknor-jackson-jsonSchema/pull/60

Included in release 1.0.28
